runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -339,8 +339,13 @@ void J9::RecognizedCallTransformer::process_jdk_internal_util_ArraysSupport_vect
       TR::Node::create(node, TR::lxor, 2, mask, TR::Node::lconst(node, -1)));
 
    TR::Node* mismatchByteIndex = TR::Node::create(node, TR::arraycmplen, 3);
-   // TODO: replace the following aladd's with generateDataAddrLoadTrees when off-heap memory changes come in
-   // See OpenJ9 issue #16717 https://github.com/eclipse-openj9/openj9/issues/16717
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
```

Comment:
`a` and `b` are children nodes of the call to vectorizedMismatch, which have multiple arguments. It doesn't seem possible to me that the call node won't have the appropriate children for the call. But I can add an extra check?

```
TR::Node* a = node->getChild(0);
TR::Node* b = node->getChild(2);
```

Reply:
misunderstanding the comment?
This is the argument description of that call:
```
     * @param a the first array to be tested for mismatch, or {@code null} for
     * direct memory access
     * @param aOffset the relative offset, in bytes, from the base address of
     * the first array to test from, otherwise if the first array is
     * {@code null}, an absolute address pointing to the first element to test.
     * @param b the second array to be tested for mismatch, or {@code null} for
     * direct memory access
```

if a or b is NULL, as it is, your dataaddr load will crash?

Reply:
I see, you mean the value not the pointer. 
Hmm... that might need something similar to Unsafe.copyMemory of a runtime check... also although against the standard there seem to be nothing stopping calling this with a non-array object as the isArray checks are commented out from OpenJDK:
```
        // assert a.getClass().isArray();
        // assert b.getClass().isArray();
```